pallets/liquidity-pools/src/lib.rs

```diff
@@ -131,7 +132,7 @@ pub mod pallet {
 	pub struct Pallet<T>(_);
 
 	#[pallet::config]
-	pub trait Config: frame_system::Config {
+	pub trait Config: frame_system::Config<AccountId = AccountId32> {
```

Comment:
Everything works fine in IT except for a compilation error with the fudge environment. I bet you, guys, to give it a try... I had never experienced it before.

Reply:
And I assume you restricted `AccountId` to `AccountId32` so that you can easily call `domain_address.account()`.

Reply:
But I think `domain_address.account()` should return `AccountId` instead of `AccountId32`.

Reply:
> And I assume you restricted AccountId to AccountId32 so that you can easily call domain_address.account()

Yes, anyways, we were using `From<[u8; 32]>` and `Into<[u8; 32]>` as a restriction in that pallet.

> But I think domain_address.account() should return AccountId instead of AccountId32.

`AccountId` is only a type alias for `AccountId32`, which should not affect it. Ideally, `AccountId` is (should be) a runtime definition. I used the real type `AccountId32`, just to be more close to the baseline

Reply:
```rust
fn foo<T: Runtime + FudgeSupport>(f: ApiRefOf<T>) {} // compiles
```

```rust
fn bar<T: Runtime + FudgeSupport, F: FnOnce(ApiRefOf<T>)>(f: F) {} //no compiles
```

I'm reaching the boundaries of the rust compiler? 😲

Reply:
Finally, I fixed getting the type in another, more simpler, way